x-pack/plugins/cross_cluster_replication/server/routes/api/register_auto_follow_pattern_routes.test.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { callWithRequestFactory } from '../../lib/call_with_request_factory';
+import { registerAutoFollowPatternRoutes } from './register_auto_follow_pattern_routes';
+
+jest.mock('../../lib/call_with_request_factory', () => ({ callWithRequestFactory: jest.fn() }));
+jest.mock('../../lib/is_es_error_factory', () => ({ isEsErrorFactory: () => () => true }));
+jest.mock('../../lib/license_pre_routing_factory', () => ({ licensePreRoutingFactory: () => null }));
+
+const setHttpRequestResponse = (err, response) => {
+  if (err) {
+    return callWithRequestFactory.mockReturnValueOnce(() => {
+      throw err;
+    });
+  }
+
+  callWithRequestFactory.mockReturnValueOnce(() => response);
+};
+
+describe('[API Routes] Auto Follow Pattern', () => {
+  let server;
+  let routeHandler;
+
+  beforeEach(() => {
+    server = {
+      route({ handler }) {
+        routeHandler = handler;
+      },
+    };
+  });
+
+  describe('list()', () => {
+    it('should forward the response from Elasticsearch', async () => {
+      const mock = { foo: 'bar' };
+      setHttpRequestResponse(null, mock);
+
+      registerAutoFollowPatternRoutes(server);
```

Comment:
The function we're testing is `registerAutoFollowPatternRoutes`, yet the test description and assertion both correspond with the `GET /auto_follow_patterns` endpoint. I'm having some cognitive dissonance because of these differences. It seems like we should rename the function/filename to reflect this route (e.g. import `registerGetAutoFollowPatternsRoute` from `get_auto_follow_patterns.js`). Then we could add an additional test file per route. What do you think?

Reply:
I understand your point of view, let me see how it could be better named. Here I followed what you did in _rollup_  (got the best of both rollup and index management for different things 😄 ). If the crud is relatively simple I'd stick with one file. So if it's ok with you, for now, we could leave it like this and we can split it later if it gets too big.